edns.go

```diff
@@ -102,15 +102,13 @@ func (rr *OPT) SetVersion(v uint8) {
 
 // ExtendedRcode returns the EDNS extended RCODE field (the upper 8 bits of the TTL).
 func (rr *OPT) ExtendedRcode() int {
-	return int(rr.Hdr.Ttl&0xFF000000>>24) + 15
+	return int(rr.Hdr.Ttl&0xFF000000>>24) << 4
 }
 
 // SetExtendedRcode sets the EDNS extended RCODE field.
-func (rr *OPT) SetExtendedRcode(v uint8) {
-	if v < RcodeBadVers { // Smaller than 16.. Use the 4 bits you have!
-		return
-	}
-	rr.Hdr.Ttl = rr.Hdr.Ttl&0x00FFFFFF | uint32(v-15)<<24
+// If the RCODE is not an extended RCODE, will reset the extended RCODE field to 0.
```

Comment:
I am not sure I understand. This is not what is being done in the rest of the file either.

Reply:
Sorry I wasn't clear, what I mean is this:
```
// SetExtendedRcode sets the EDNS extended RCODE field.
//
// If the RCODE is not an extended RCODE, will reset the extended RCODE field to 0. 
```

Reply:
ok, this is fixed now. I had understood what you meant correctly actually, I was just not sure because the rest of the file do not use this convention.
Anyhow, done now :D